improver/cli/apply_emos_coefficients.py

```diff
@@ -189,27 +176,13 @@ def process(cube: cli.inputcube,
     # If percentiles, re-sample percentiles and then re-badge.
     # If probabilities, generate percentiles and then re-badge.
     if input_forecast_type in ["percentiles", "probabilities"]:
-        if not realizations_count:
-            raise ValueError(
-                "The current forecast has been provided as {0}. "
-                "These {0} need to be converted to realizations "
-                "for ensemble calibration. The realizations_count "
-                "argument is used to define the number of realizations "
-                "to construct from the input {0}, so if the "
-                "current forecast is provided as {0} then "
-                "realizations_count must be defined.".format(
-                    input_forecast_type))
-        current_forecast = conversion_plugin.process(
-            current_forecast, no_of_percentiles=realizations_count)
+        current_forecast = conversion_plugin.process(current_forecast)
         current_forecast = (
             RebadgePercentilesAsRealizations().process(current_forecast))
 
     # Default number of ensemble realizations is the number in
     # the raw forecast.
-    if not realizations_count:
-        realizations_count = len(
-            current_forecast.coord('realization').points)
-
+    no_of_percentiles = len(current_forecast.coord('realization').points)
```

Comment:
This variable isn't used until line 216, and only within that scope.  Please define variable within scope.